src/main/java/com/rybka/service/CSVExportService.java

```diff
@@ -0,0 +1,41 @@
+package com.rybka.service;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import com.rybka.config.PropertyInfo;
+import com.rybka.exception.ExportFailureException;
+import com.rybka.model.CurrencyHistory;
+import lombok.extern.log4j.Log4j;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+@Log4j
+public class CSVExportService implements ExportService {
+    private final CsvMapper csvMapper;
+
+    public CSVExportService(CsvMapper csvMapper) {
+        this.csvMapper = csvMapper;
+    }
+
+    public void export(List<CurrencyHistory> histories) {
+        csvMapper.disable(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY);
+        CsvSchema csvSchema = csvMapper.schemaFor(CurrencyHistory.class).withHeader();
+        ObjectWriter writer = csvMapper.writer(csvSchema.withLineSeparator("\n"));
+
+        try (var fileWriter = new FileWriter(PropertyInfo.RESOURCE_PATH
+                + new RandomNameService().generateFileName()
```

Comment:
create method, there isn't any need to create a hole service. 
at least you can create FileUntils.class with static method.

Reply:
fixed